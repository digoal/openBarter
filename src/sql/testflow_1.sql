drop extension if exists flowf cascade;
create extension flowf;

RESET client_min_messages;
RESET log_error_verbosity;
SET client_min_messages = notice;
SET log_error_verbosity = terse;

SELECT '[(1,1,2,3,4,5,6,7.00)]'::yflow;
SELECT yflow_init(ROW(1,1,1,1,100,'|q1',200,'|q2',50,'((1,1),(1,1))'::box,'((1,1),(1,1))'::box,1,'((1,1),(1,1))'::box)::yorder);

SELECT yflow_grow_backward(
    ROW(1,2,1,2,100,'|q2',200,'|q1',50,'((1,1),(1,1))'::box,'((1,1),(1,1))'::box,1,'((1,1),(1,1))'::box)::yorder, 
    ROW(1,1,1,1,100,'|q1',200,'|q2',50,'((1,1),(1,1))'::box,'((1,1),(1,1))'::box,1,'((1,1),(1,1))'::box)::yorder, 
yflow_init(
    ROW(1,1,1,1,100,'|q1',200,'|q2',50,'((1,1),(1,1))'::box,'((1,1),(1,1))'::box,1,'((1,1),(1,1))'::box)::yorder) );

SELECT yflow_finish(    ROW(1,2,1,2,100,'|q2',200,'|q1',50,'((1,1),(1,1))'::box,'((1,1),(1,1))'::box,1,'((1,1),(1,1))'::box)::yorder,
yflow_grow_backward(    ROW(1,2,1,2,100,'|q2',200,'|q1',50,'((1,1),(1,1))'::box,'((1,1),(1,1))'::box,1,'((1,1),(1,1))'::box)::yorder, 
                        ROW(1,1,1,1,100,'|q1',200,'|q2',50,'((1,1),(1,1))'::box,'((1,1),(1,1))'::box,1,'((1,1),(1,1))'::box)::yorder, 
yflow_init(             ROW(1,1,1,1,100,'|q1',200,'|q2',50,'((1,1),(1,1))'::box,'((1,1),(1,1))'::box,1,'((1,1),(1,1))'::box)::yorder) ),
                        ROW(1,1,1,1,100,'|q1',200,'|q2',50,'((1,1),(1,1))'::box,'((1,1),(1,1))'::box,1,'((1,1),(1,1))'::box)::yorder);

SELECT yflow_dim(yflow_init(ROW(1,1,1,2,100,'|q1',200,'|q2',50,'((1,1),(1,1))'::box,'((1,1),(1,1))'::box,1,'((1,1),(1,1))'::box)::yorder));

SELECT yflow_contains_oid(1,yflow_init(ROW(1,1,1,2,100,'|q1',200,'|q2',50,'((1,1),(1,1))'::box,'((1,1),(1,1))'::box,1,'((1,1),(1,1))'::box)::yorder));
SELECT yflow_contains_oid(2,yflow_init(ROW(1,1,1,2,100,'|q1',200,'|q2',50,'((1,1),(1,1))'::box,'((1,1),(1,1))'::box,1,'((1,1),(1,1))'::box)::yorder));


select yflow_is_draft('[(1,35, 93, 35, 21170, 2685, 2685, 1.000000),(1,636, 50, 636, 12213, 95415, 95415, 1.000000
),(1,389, 68, 389, 23785, 29283, 29283, 1.000000),(1,274, 12, 274, 58834, 80362, 80362, 1.000000),(1,12, 
55, 12, 35136, 55490, 55490, 1.000000)]'::yflow);

select yflow_reduce(x.f1,x.f1,true) from (select '[(1,35, 93, 35, 21170, 2685, 2685, 1.000000),(1,636, 50, 636, 12213, 95415, 95415, 1.000000
),(1,389, 68, 389, 23785, 29283, 29283, 1.000000),(1,274, 12, 274, 58834, 80362, 80362, 1.000000),(1,12, 
55, 12, 35136, 55490, 55490, 1.000000)]'::yflow as f1) x;

select yflow_show('[(1,35, 93, 35, 21170, 2685, 2685, 1.000000),(1,636, 50, 636, 12213, 95415, 95415, 1.000000
),(1,389, 68, 389, 23785, 29283, 29283, 1.000000),(1,274, 12, 274, 58834, 80362, 80362, 1.000000),(1,12, 
55, 12, 35136, 55490, 55490, 1.000000)]'::yflow);
select yflow_to_matrix('[(1,35, 93, 35, 21170, 2685, 2685, 1.000000),(1,636, 50, 636, 12213, 95415, 95415, 1.000000
),(1,389, 68, 389, 23785, 29283, 29283, 1.000000),(1,274, 12, 274, 58834, 80362, 80362, 1.000000),(1,12, 
55, 12, 35136, 55490, 55490, 1.000000)]'::yflow);

select yflow_show('[(1,35, 93, 35, 21170, 2685, 2685, 1.000000),(1,636, 50, 636, 12213, 95415, 95415, 1.000000
),(1,389, 68, 389, 23785, 29283, 29283, 1.000000),(1,274, 12, 274, 58834, 80362, 80362, 1.000000),(1,12, 
55, 12, 35136, 55490, 55490, 1.000000)]'::yflow);
select yflow_qtts('[(1,35, 93, 35, 21170, 2685, 2685, 1.000000),(1,636, 50, 636, 12213, 95415, 95415, 1.000000
),(1,389, 68, 389, 23785, 29283, 29283, 1.000000),(1,274, 12, 274, 58834, 80362, 80362, 1.000000),(1,12, 
55, 12, 35136, 55490, 55490, 1.000000)]'::yflow);

select yflow_show('[(1,62, 62, 6, 49210, 60487, 55111, 1.000000),(1,64, 64, 4, 64784, 55162, 53296, 1.000000),(1,52, 52, 14, 34697, 57236, 56208, 1.000000),(1,86, 86, 11, 19239, 28465, 27569, 1.000000),(1,87, 87, 4, 20786, 61473, 60554, 1.000000),(1,33, 33, 16, 828, 12515, 12515, 1.000000),(1,1, 1, 1, 35542, 66945, 17677, 1.000000),(1,102, 102, 15, 87633, 69633, 33223, 1.000000)]'::yflow);
select yflow_qtts('[(1,62, 62, 6, 49210, 60487, 55111, 1.000000),(1,64, 64, 4, 64784, 55162, 53296, 1.000000),(1,52, 52, 14, 34697, 57236, 56208, 1.000000),(1,86, 86, 11, 19239, 28465, 27569, 1.000000),(1,87, 87, 4, 20786, 61473, 60554, 1.000000),(1,33, 33, 16, 828, 12515, 12515, 1.000000),(1,1, 1, 1, 35542, 66945, 17677, 1.000000),(1,102, 102, 15, 87633, 69633, 33223, 1.000000)]'::yflow);

select yflow_show('[(2, 3298, 3298, 25, 87502, 61824, 20450, 1.000000),(2, 1090, 1090, 91, 41375, 55121, 1, 1.000000),(78, 10013, 10013, 72, 55121, 58560, 58559, 1.000000)]'::yflow);

/* IGNOREOMEGA QTTNOLIMIT */
select yflow_show('[
(2, 3298, 3298, 25, 10, 20, 20, 1.000000),
(2, 1090, 1090, 91, 10, 10, 10, 1.000000),
(78, 10013, 10013, 72, 0, 0, 0, 1.000000)]'::yflow);
