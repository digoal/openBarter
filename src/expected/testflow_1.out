create extension cube with version '1.0';
create extension hstore with version '1.1';
create extension flowf; -- with version '1.0';
RESET client_min_messages;
RESET log_error_verbosity;
SET client_min_messages = notice;
SET log_error_verbosity = terse;
SELECT '[(1,1,2,3,4,5,6,7.00)]'::yflow;
               yflow               
-----------------------------------
 [(1, 1, 2, 3, 4, 5, 6, 7.000000)]
(1 row)

SELECT yflow_init(ROW(1,1,1,1,100,'q1',200,'q2',50)::yorder);
               yflow_init                
-----------------------------------------
 [(1, 1, 1, 1, 100, 200, 50, -1.000000)]
(1 row)

SELECT yflow_grow(ROW(1,2,1,2,100,'q2',200,'q1',50)::yorder, ROW(1,1,1,1,100,'q1',200,'q2',50)::yorder, yflow_init(ROW(1,1,1,1,100,'q1',200,'q2',50)::yorder) );
                                  yflow_grow                                  
------------------------------------------------------------------------------
 [(1, 2, 2, 1, 100, 200, 50, 1.000000),(1, 1, 1, 1, 100, 200, 50, -1.000000)]
(1 row)

SELECT yflow_finish(ROW(1,2,1,2,100,'q2',200,'q1',50)::yorder,
yflow_grow(ROW(1,2,1,2,100,'q2',200,'q1',50)::yorder, ROW(1,1,1,1,100,'q1',200,'q2',50)::yorder, yflow_init(ROW(1,1,1,1,100,'q1',200,'q2',50)::yorder) ),
ROW(1,1,1,1,100,'q1',200,'q2',50)::yorder);
                                yflow_finish                                 
-----------------------------------------------------------------------------
 [(1, 2, 2, 1, 100, 200, 50, 1.000000),(1, 1, 1, 1, 100, 200, 50, 1.000000)]
(1 row)

SELECT yflow_dim(yflow_init(ROW(1,1,1,2,100,'q1',200,'q2',50)::yorder));
 yflow_dim 
-----------
         1
(1 row)

SELECT yflow_contains_oid(1,yflow_init(ROW(1,1,1,2,100,'q1',200,'q2',50)::yorder));
 yflow_contains_oid 
--------------------
 f
(1 row)

SELECT yflow_contains_oid(2,yflow_init(ROW(1,1,1,2,100,'q1',200,'q2',50)::yorder));
 yflow_contains_oid 
--------------------
 t
(1 row)

select yflow_is_draft('[(1,35, 93, 35, 21170, 2685, 2685, 1.000000),(1,636, 50, 636, 12213, 95415, 95415, 1.000000
),(1,389, 68, 389, 23785, 29283, 29283, 1.000000),(1,274, 12, 274, 58834, 80362, 80362, 1.000000),(1,12, 
55, 12, 35136, 55490, 55490, 1.000000)]'::yflow);
 yflow_is_draft 
----------------
 t
(1 row)

select yflow_reduce(x.f1,x.f1,true) from (select '[(1,35, 93, 35, 21170, 2685, 2685, 1.000000),(1,636, 50, 636, 12213, 95415, 95415, 1.000000
),(1,389, 68, 389, 23785, 29283, 29283, 1.000000),(1,274, 12, 274, 58834, 80362, 80362, 1.000000),(1,12, 
55, 12, 35136, 55490, 55490, 1.000000)]'::yflow as f1) x;
                                                                                                                 yflow_reduce                                                                                                                  
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 [(1, 35, 93, 35, 21170, 2685, 0, 1.000000),(1, 636, 50, 636, 12213, 95415, 78129, 1.000000),(1, 389, 68, 389, 23785, 29283, 11746, 1.000000),(1, 274, 12, 274, 58834, 80362, 60622, 1.000000),(1, 12, 55, 12, 35136, 55490, 29800, 1.000000)]
(1 row)

select yflow_show('[(1,35, 93, 35, 21170, 2685, 2685, 1.000000),(1,636, 50, 636, 12213, 95415, 95415, 1.000000
),(1,389, 68, 389, 23785, 29283, 29283, 1.000000),(1,274, 12, 274, 58834, 80362, 80362, 1.000000),(1,12, 
55, 12, 35136, 55490, 55490, 1.000000)]'::yflow);
                                                                                                                                        yflow_show                                                                                                                                        
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 YFLOW [(1, 35, 93, 35, 21170, 2685, 2685, 1.000000 :2685),(1, 636, 50, 636, 12213, 95415, 95415, 1.000000 :17286),(1, 389, 68, 389, 23785, 29283, 29283, 1.000000 :17537),(1, 274, 12, 274, 58834, 80362, 80362, 1.000000 :19740),(1, 12, 55, 12, 35136, 55490, 55490, 1.000000 :25690)]+
 
(1 row)

select yflow_to_matrix('[(1,35, 93, 35, 21170, 2685, 2685, 1.000000),(1,636, 50, 636, 12213, 95415, 95415, 1.000000
),(1,389, 68, 389, 23785, 29283, 29283, 1.000000),(1,274, 12, 274, 58834, 80362, 80362, 1.000000),(1,12, 
55, 12, 35136, 55490, 55490, 1.000000)]'::yflow);
                                                                                   yflow_to_matrix                                                                                   
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {{35,35,93,21170,2685,2685,2685},{636,636,50,12213,95415,95415,17286},{389,389,68,23785,29283,29283,17537},{274,274,12,58834,80362,80362,19740},{12,12,55,35136,55490,55490,25690}}
(1 row)

select yflow_show('[(1,35, 93, 35, 21170, 2685, 2685, 1.000000),(1,636, 50, 636, 12213, 95415, 95415, 1.000000
),(1,389, 68, 389, 23785, 29283, 29283, 1.000000),(1,274, 12, 274, 58834, 80362, 80362, 1.000000),(1,12, 
55, 12, 35136, 55490, 55490, 1.000000)]'::yflow);
                                                                                                                                        yflow_show                                                                                                                                        
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 YFLOW [(1, 35, 93, 35, 21170, 2685, 2685, 1.000000 :2685),(1, 636, 50, 636, 12213, 95415, 95415, 1.000000 :17286),(1, 389, 68, 389, 23785, 29283, 29283, 1.000000 :17537),(1, 274, 12, 274, 58834, 80362, 80362, 1.000000 :19740),(1, 12, 55, 12, 35136, 55490, 55490, 1.000000 :25690)]+
 
(1 row)

select yflow_qtts('[(1,35, 93, 35, 21170, 2685, 2685, 1.000000),(1,636, 50, 636, 12213, 95415, 95415, 1.000000
),(1,389, 68, 389, 23785, 29283, 29283, 1.000000),(1,274, 12, 274, 58834, 80362, 80362, 1.000000),(1,12, 
55, 12, 35136, 55490, 55490, 1.000000)]'::yflow);
           yflow_qtts            
---------------------------------
 {19740,25690,35136,55490,55490}
(1 row)

select yflow_show('[(1,62, 62, 6, 49210, 60487, 55111, 1.000000),(1,64, 64, 4, 64784, 55162, 53296, 1.000000),(1,52, 52, 14, 34697, 57236, 56208, 1.000000),(1,86, 86, 11, 19239, 28465, 27569, 1.000000),(1,87, 87, 4, 20786, 61473, 60554, 1.000000),(1,33, 33, 16, 828, 12515, 12515, 1.000000),(1,1, 1, 1, 35542, 66945, 17677, 1.000000),(1,102, 102, 15, 87633, 69633, 33223, 1.000000)]'::yflow);
                                                                                                                                                                                                                 yflow_show                                                                                                                                                                                                                 
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 YFLOW [(1, 62, 62, 6, 49210, 60487, 55111, 1.000000 :2543),(1, 64, 64, 4, 64784, 55162, 53296, 1.000000 :1500),(1, 52, 52, 14, 34697, 57236, 56208, 1.000000 :1187),(1, 86, 86, 11, 19239, 28465, 27569, 1.000000 :843),(1, 87, 87, 4, 20786, 61473, 60554, 1.000000 :1726),(1, 33, 33, 16, 828, 12515, 12515, 1.000000 :12515),(1, 1, 1, 1, 35542, 66945, 17677, 1.000000 :11310),(1, 102, 102, 15, 87633, 69633, 33223, 1.000000 :4312)]+
 
(1 row)

select yflow_qtts('[(1,62, 62, 6, 49210, 60487, 55111, 1.000000),(1,64, 64, 4, 64784, 55162, 53296, 1.000000),(1,52, 52, 14, 34697, 57236, 56208, 1.000000),(1,86, 86, 11, 19239, 28465, 27569, 1.000000),(1,87, 87, 4, 20786, 61473, 60554, 1.000000),(1,33, 33, 16, 828, 12515, 12515, 1.000000),(1,1, 1, 1, 35542, 66945, 17677, 1.000000),(1,102, 102, 15, 87633, 69633, 33223, 1.000000)]'::yflow);
           yflow_qtts           
--------------------------------
 {11310,4312,87633,69633,33223}
(1 row)

select yflow_show('[(2, 3298, 3298, 25, 87502, 61824, 20450, 1.000000),(2, 1090, 1090, 91, 41375, 55121, 1, 1.000000),(78, 10013, 10013, 72, 55121, 58560, 58559, 1.000000)]'::yflow);
                                                                                yflow_show                                                                                
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 YFLOW [(2, 3298, 3298, 25, 87502, 61824, 20450, 1.000000 :-1),(2, 1090, 1090, 91, 41375, 55121, 1, 1.000000 :-1),(78, 10013, 10013, 72, 55121, 58560, -1, 1.000000 :-1)]+
 
(1 row)

